File: Leetcode July Challenge/Gray Code.cpp
class Solution {
public:
    vector<int> grayCode(int n) {
        vector<string> garyString = generate(n);
        vector<int> ans;
        for(auto i:garyString)
        {
            ans.push_back(stoi(i,0,2));
        }
        return ans;
    }
    
    vector<string> generate(int n)
    {
        if(n==1)
        {
            return {"0","1"};
        }
        vector<string> ans;
        vector<string> temp = generate(n-1);
        for(int i=0;i<temp.size();i++)
        {
            ans.push_back("0"+temp[i]);
        }
        for(int i=temp.size()-1;i>=0;i--)
        {
            ans.push_back("1"+temp[i]);
        }
        return ans;
    }
    
};

File: Leetcode July Challenge/Python/Gray Code.py
class Solution:
    
    
    def grayCode(self, n: int) -> List[int]:
        
        def helper(n):
            
            if(n==1):
                return ["0","1"]
        
            x=helper(n-1)
            res=[]
            print(n, x)
            for i in range(len(x)):
                s=x[i]
                res.append("0"+s)
            print(n, res)    
            for i in range(len(x)-1,-1,-1):
                s=x[i]
                res.append("1"+s)
            print(n, res)
            return res
        
        res=helper(n)
        fres=[]
        for i in res:
            fres.append(int(i,2))
        return fres                        
            
